docs/admin/healthcheck.md

```diff
@@ -0,0 +1,174 @@
+  misconfigured reverse proxy)
+
+**Solution:** Inspect the `HealthzResponse` in the health check output. This
+should give you a good indication of the root cause.
+
+## Database
+
+Coder continuously executes a short database query to validate that it can reach its configured database, and also
+measures the median latency over 5 attempts.
+
+### <a name="EDB01">EDB01: Database Ping Failed</a>
+
+**Problem:** This error code is returned if any attempt to execute this database query fails.
+
+**Solution:** Investigate the health of the database.
+
+### <a name="EDB02">EDB02: Database Ping High</a>
+
+**Problem:** This code is returned if the median latency is higher than the [configured threshold](../cli/server.md#--health-check-threshold-database).
+This may not be an error as such, but is an indication of a potential issue.
+
+**Solution:** Investigate the sizing of the configured database with regard to Coder's current activity and usage. It
+may be necessary to increase the resources allocated to Coder's database. Alternatively, you can raise the configured
+threshold to a higher value (this will not address the root cause).
+
+> [!TIP]
+> - You can enable [detailed database metrics](../cli/server.md#--prometheus-collect-db-metrics) in Coder's
+> Prometheus endpoint.
+> - Fif you have [tracing enabled](../cli/server.md#--trace), these traces may also contain useful information regarding
+> Coder's database activity.
+
+## DERP
+
+Coder workspace agents may use [DERP (Designated Encrypted Relay for Packets)](https://tailscale.com/blog/how-tailscale-works/#encrypted-tcp-relays-derp) to communicate with Coder.
+This requires connectivity to a number of configured [DERP servers](../cli/server.md#--derp-config-path) which are used
+to relay traffic between Coder and workspace agents. Coder periodically queries the health of its configured DERP servers and may return one or more of the following:
+
+### <a name="EDERP01">EDERP01: DERP Node Uses Websocket</a>
+
+**Problem:** When Coder attempts to establish a connection to one or more DERP servers, it sends a specific `Upgrade: derp` HTTP header.
+Some load balancers may block this header, in which case Coder will fall back to `Upgrade: websocket`.
+This is not necessarily a fatal error, but a possible indication of a misconfigured reverse HTTP proxy.
+
+> [!NOTE]
+> This may also be shown if you have [forced websocket connections for DERP](../cli/server.md#--derp-force-websockets).
+
+**Solution:** ensure that any configured reverse proxy does not strip the `Upgrade: derp` header.
+
+### <a name="EDERP02">EDERP02: One or more DERP nodes unhealthy</a>
+
+**Problem:** This s shown if Coder is unable to reach one or more configured DERP servers. Clients will fall back to use the remaining
+DERP servers, but performance may be impacted for clients closest to the unhealthy DERP server.
+
+**Solution:** Ensure that the DERP server is available and reachable over the network on port 443.
```

Comment:
Any curl here?

Reply:
I'm not sure of the right command here, so I'll just do a curl for now.